### PR TITLE
Refactor/gui editbox

### PR DESCRIPTION
**Pull request by nerzhul**
Add compact, short information about your PR for easier understanding:

- Continue work in order to dedup code duplication between 2 editboxes classes

## To do

This PR is Ready for Review.

## How to test
Run MT and play with boxes

### CODE REVIEW

File: src/gui/guiEditBox.cpp [hunk 1]
```diff
@@ -93,3 +101,733 @@ void GUIEditBox::setAutoScroll(bool enable)
...
+									       [lineNo + 1] +
+							       core::max_((u32)1,
+									       m_broken_text[lineNo +
+											       1]
+											       .size()) -
+							       1;
+					else
+						m_cursor_pos = m_broken_text_positions
+									       [lineNo + 1] +
+							       cp;
```
Comment (SmallJoker) [inline src/gui/guiEditBox.cpp:552]: What's this formatting? I'd prefer to keep the old style with `// clang-format off` "guards" than this. Same above.

File: src/gui/guiEditBox.cpp [hunk 2]
```diff
@@ -93,3 +101,733 @@ void GUIEditBox::setAutoScroll(bool enable)
...
+	} else {
+		if (m_vscrollbar->isVisible()) {
+			m_vscrollbar->setVisible(false);
+			m_vscroll_pos = 0;
+			m_vscrollbar->setPos(0);
+			m_vscrollbar->setMax(1);
+			m_vscrollbar->setPageSize(s32(getTextDimension().Height));
+		}
+	}
+}
```
Comment (SmallJoker) [inline src/gui/guiEditBox.cpp:833]: \+ \n

**Review by SmallJoker — COMMENTED** [review]
Works, but the code style is not visually appealing.

File: src/gui/guiEditBox.cpp (hunk 1 above)
Comment (nerzhul) [reply src/gui/guiEditBox.cpp:552]: didn't saw that crap omg :)

Comment (nerzhul) [reply src/gui/guiEditBox.cpp:552]: maybe too many indent i may use a function instead i think

**Comment by nerzhul** [comment]
Comments are now fixed @SmallJoker it's a second step in order to merge GUIEditBox codes.
After merging, you talked about backporting to irrlicht. I agree with you that we should integrate this object inside irrlicht library, and let library handle all of this properly, it means also if we use the forked irrlicht version we can also do better SDL2 integration into these objects, as we will need to add a new event for typed text to be handled here directly

File: src/gui/guiEditBox.cpp [hunk 3]
```diff
@@ -93,3 +101,727 @@ void GUIEditBox::setAutoScroll(bool enable)
...
+			core::stringw s = Text.subString(0, m_cursor_pos);
+			s.append(p);
+			s.append(Text.subString(
+					m_cursor_pos, Text.size() - m_cursor_pos));
+
+			if (!m_max || s.size() <= m_max) // thx to
+							 // Fish
+							 // FH for
+							 // fix
+			{
```
Comment (SmallJoker) [inline src/gui/guiEditBox.cpp:597]:
```suggestion
			if (!m_max || s.size() <= m_max) {
```

Same below as well.

Comment (nerzhul) [reply src/gui/guiEditBox.cpp:597]: ah strange that the formatter didn't fixed it

**Review by SmallJoker — APPROVED** [review]

**Pull request merged**